src/main/java/uk/gov/hmcts/reform/ethos/ecm/consumer/service/SingleUpdateService.java

```diff
@@ -74,4 +83,42 @@ private CCDRequest getReturnedRequest(String accessToken, String caseTypeId, Str
         }
     }
 
+    private void updateMultipleReferenceLinkMarkUp(SubmitEvent submitEvent, String accessToken,
+                                                   UpdateCaseMsg updateCaseMsg) throws IOException {
+
+        if (isNullOrEmpty(submitEvent.getCaseData().getMultipleReferenceLinkMarkUp())) {
+
+            List<SubmitMultipleEvent> submitMultipleEvents = retrieveMultipleCase(accessToken, updateCaseMsg);
+
+            if (!submitMultipleEvents.isEmpty()) {
+
+                submitEvent.getCaseData().setMultipleReferenceLinkMarkUp(
+                    generateMarkUp(ccdGatewayBaseUrl,
+                                   String.valueOf(submitMultipleEvents.get(0).getCaseId()),
+                                   submitEvent.getCaseData().getMultipleReference()));
+
+            }
+
+        }
+
+    }
+
+    private List<SubmitMultipleEvent> retrieveMultipleCase(String authToken,
+                                                           UpdateCaseMsg updateCaseMsg) throws IOException {
+
+        return ccdClient.retrieveMultipleCasesElasticSearchWithRetries(
+            authToken,
+            updateCaseMsg.getCaseTypeId(),
+            updateCaseMsg.getMultipleRef());
+
+    }
+
+    private String generateMarkUp(String ccdGatewayBaseUrl, String caseId, String ethosCaseRef) {
```

Comment:
Only one minor thing. There are a lot of blank lines that aren't necessary - this method has more blank lines that lines of code. You only need to add blank lines if they make your code more readable.